examples/driver-websockets/src/Log.purs

```diff
@@ -21,13 +31,35 @@ component =
   where
 
   initialState :: State
-  initialState = { messages: [] }
+  initialState = { messages: [] , inputText: "" }
 
   render :: State -> H.ComponentHTML Query
   render state =
-    HH.ol_ $ map (\msg -> HH.li_ [ HH.text msg ]) state.messages
+    HH.div_
+    [ HH.ol_ $ map (\msg -> HH.li_ [ HH.text msg ]) state.messages
+    , HH.input
+      [ HP.type_ HP.InputText
+      , HP.value (state.inputText)
+      , HE.onValueInput (HE.input UpdateInputText)
+      ]
```

Comment:
Please double-indent the nested arrays for properties / child elements throughout.